packages/cli/src/UserManagement/routes/users.ts

```diff
@@ -203,4 +193,140 @@ export function addUsersMethods(this: N8nApp): void {
+					{ user: deletedUser },
+					{ user: transferUser },
+				);
+			} else {
+				const ownedWorkflows = await Db.collections.SharedWorkflow!.find({
+					relations: ['workflow'],
+					where: { user: deletedUser },
+				});
+				if (ownedWorkflows.length) {
+					await Db.collections.Workflow!.delete({
+						id: In(ownedWorkflows.map((ownedWorkflow) => ownedWorkflow.workflow.id)),
+					});
+					await Db.collections.SharedWorkflow!.delete({ user: deletedUser });
+				}
+
+				const ownedCredentials = await Db.collections.SharedCredentials!.find({
+					relations: ['credentials'],
+					where: { user: deletedUser },
+				});
+				if (ownedCredentials.length) {
+					await Db.collections.Credentials!.delete({
+						id: In(ownedCredentials.map((ownedCredential) => ownedCredential.credentials.id)),
+					});
+					await Db.collections.SharedWorkflow!.delete({ user: deletedUser });
+				}
+
+				const queryBuilderCredentials = Db.collections.Credentials!.createQueryBuilder('c');
+				queryBuilderCredentials.innerJoin('c.shared', 'shared');
+				queryBuilderCredentials.andWhere('shared.userId = :userId', {
+					userId: deletedUser.id,
+				});
+
+				queryBuilderCredentials.select(['c.id']);
+
+				const credentials =
+					(await queryBuilderCredentials.getMany()) as unknown as ICredentialsResponse[];
+
+				if (credentials.length) {
+					await Db.collections.Credentials!.remove(credentials);
+					await Db.collections.SharedCredentials!.delete({ user: deletedUser });
+				}
+			}
+			await Db.collections.User!.delete({ id: deletedUser.id });
+			return { success: true };
+		}),
+	);
+
+	this.app.post(
+		`/${this.restEndpoint}/users/:id/reinvite`,
+		ResponseHelper.send(async (req: Request) => {
+			if (!isEmailSetup()) {
+				throw new ResponseHelper.ResponseError(
+					'Email sending must be set up in order to invite other users',
+					undefined,
+					500,
+				);
+			}
+
+			const user = await Db.collections.User!.findOne({ id: req.params.id });
+
+			if (!user) {
+				throw new ResponseHelper.ResponseError('User not found', undefined, 404);
+			}
+
+			if (user.password) {
+				throw new ResponseHelper.ResponseError(
+					'User has already accepted the invite',
+					undefined,
+					400,
+				);
+			}
+
+			let inviteAcceptUrl = GenericHelpers.getBaseUrl();
+			const domain = inviteAcceptUrl;
+			if (!inviteAcceptUrl.endsWith('/')) {
+				inviteAcceptUrl += '/';
+			}
+			// eslint-disable-next-line @typescript-eslint/restrict-template-expressions, @typescript-eslint/no-unsafe-member-access
+			inviteAcceptUrl += `signup/inviterId=${(req.user as User).id}&inviteeId${user.id}`;
+
+			const mailer = getInstance();
+			// eslint-disable-next-line no-await-in-loop
+			const result = await mailer.invite({
+				email: user.email,
+				inviteAcceptUrl,
+				domain,
+			});
+
+			if (result.success) {
+				throw new ResponseHelper.ResponseError('Unable to send email', undefined, 500);
+			}
+			return { success: false };
```

Comment:
not knowing what the result looks like looks the logic off 🙈

Reply:
Ops, my bad!